components/dialog/README.md

```diff
@@ -258,6 +258,80 @@ document.querySelector('#open').addEventListener('click', () => {
 });
 ```
 
+## Async Dialogs
```

Comment:
Made this a level 2 section rather than putting it under general since it is available in fullscreen. Could also put it under general then point at it from fullscreen.